src/Http/Middleware/AuthenticateSession.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Laravel\Sanctum\Http\Middleware;
+
+use Closure;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Auth\SessionGuard;
+use Illuminate\Contracts\Auth\Factory as AuthFactory;
+use Illuminate\Http\Request;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuthenticateSession
+{
+    /**
+     * The authentication factory implementation.
+     *
+     * @var \Illuminate\Contracts\Auth\Factory
+     */
+    protected $auth;
+
+    /**
+     * Create a new middleware instance.
+     *
+     * @param  \Illuminate\Contracts\Auth\Factory  $auth
+     * @return void
+     */
+    public function __construct(AuthFactory $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Closure(\Illuminate\Http\Request): (\Symfony\Component\HttpFoundation\Response)  $next
+     *
+     * @throws \Illuminate\Auth\AuthenticationException
+     */
+    public function handle(Request $request, Closure $next): Response
+    {
+        if (! $request->hasSession() || ! $request->user()) {
+            return $next($request);
+        }
+
+        $guards = Collection::make(Arr::wrap(config('sanctum.guard')))
+            ->mapWithKeys(fn ($guard) => [$guard => $this->auth->guard($guard)])
+            ->filter(fn ($guard) => $guard instanceof SessionGuard);
+
+        if ($request->hasSession()) {
```

Comment:
The response logic is different here.

Reply:
`$request->hasSession()` will always be true, otherwise it would return on line 44. It can be removed.

Reply:
This would caused an issue if `$request->user()` is not `null` while `$request->session()` is `null`.

![CleanShot 2023-08-23 at 18 58 06](https://github.com/laravel/sanctum/assets/172966/ddbcc0df-ceb3-4030-8363-8e99393667c6)

Reply:
I think we may be talking about different things 😄 

If `$request->hasSession()` is false, the condition on line 43 is true and the method will return on line 44. Line 46+ can only be reached when `$request->hasSession()` is true.

![image](https://github.com/laravel/sanctum/assets/571773/477282b1-2f62-47a4-814f-b36993cb14a4)